recipes/thrift/all/conanfile.py

```diff
@@ -130,17 +151,53 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
-        # Make sure libs are link in correct order. Important thing is that libthrift/thrift is last
-        # (a little naive to sort, but libthrift/thrift should end up last since rest of the libs extend it with an abbrevation: 'thriftnb', 'thriftz')
-        # The library that needs symbols must be first, then the library that resolves the symbols should come after.
-        self.cpp_info.libs.sort(reverse = True)
-
+        libsuffix = "{}{}".format(
+            str(self.settings.compiler.runtime).lower()[:2] if self.settings.compiler == "Visual Studio" else "",
+            "d" if self.settings.build_type == "Debug" else ""
+        )
+
+        self.cpp_info.filenames["cmake_find_package"] = "Thrift"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "Thrift"
+        self.cpp_info.names["cmake_find_package"] = "thrift"
+        self.cpp_info.names["cmake_find_package_multi"] = "thrift"
+
+        self.cpp_info.components["libthrift"].libs = ["thrift" + libsuffix]
+        self.cpp_info.components["libthrift"].names["cmake_find_package"] = "thrift"
+        self.cpp_info.components["libthrift"].names["cmake_find_package_multi"] = "thrift"
+        self.cpp_info.components["libthrift"].names["pkg_config"] = "thrift"
         if self.settings.os == "Windows":
-            # To avoid error C2589: '(' : illegal token on right side of '::'
-            self.cpp_info.defines.append("NOMINMAX")
+            # To avoid error C2589: "(" : illegal token on right side of "::"
+            self.cpp_info.components["libthrift"].defines.append("NOMINMAX")
         elif self.settings.os == "Linux":
-            self.cpp_info.system_libs.extend(["m", "pthread"])
+            self.cpp_info.components["libthrift"].system_libs.extend(["m", "pthread"])
+        self.cpp_info.components["libthrift"].requires.append("boost::boost")
+        if self.options.with_openssl:
+            self.cpp_info.components["libthrift"].requires.append("openssl::openssl")
+        if self.options.with_libevent:
+            self.cpp_info.components["libthrift"].requires.append("libevent::libevent")
+
+        # FIXME: this generates thrift::thriftz, but should be thriftz::thriftz
```

Comment:
Is there an issue for this in the conan repo?

Reply:
https://github.com/conan-io/conan/issues/7615
More or less.